### ISSUE TITLE
Invalid Headers.js syntax ?

### ISSUE THREAD

**Issue by danielthedifficult** [mswjs/headers-polyfill#36]
Following the recent 3.0.9, I'm having this issue when trying to start storybook (that does have msw/msw-storybook-addon :

```
ERROR in /project/node_modules/headers-polyfill/lib/Headers.js 9:25
Module parse failed: Unexpected token (9:25)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| class HeadersPolyfill {
|     // Normalized header {"name":"a, b"} storage.
>     [NORMALIZED_HEADERS] = {};
|     // Keeps the mapping between the raw header name
|     // and the normalized header name to ease the lookup.
```

[Found someone else on S/O with the same issue as well.](https://stackoverflow.com/questions/72922797/react-js-module-parse-failed-unexpected-token)

Reverting to 3.0.4 via yarn resolutions worked for me.

FWIW I'm using babel-loader 8.1.0 in case that makes a difference, as the loader was mentioned.

Also, I'm seeing that Headers.js for 3.0.4 looks like minified js, whereas this looks much more like plain (pre-transpiler) javascript.

I do see the change in #33 that switches targets from ES5 to esnext... would that have this effect?

Please let me know if you need any other information about my project.
node v16.16

**Comment by abienz** [comment]
Yes I'm noticing the same problem, I don't use yarn however and don't have the option to use a 'resolutions' option, this effectively stops my development using MSW with Storybook

**Comment by danielthedifficult** [comment]
@abienz YMMV, but [this is a third-party package for NPM that accomplishes the same thing.](https://www.npmjs.com/package/npm-force-resolutions)

**Comment by abienz** [comment]
Thank you @danielthedifficult , I did notice there were packages that do this, but my project has strict rules around 3rd party dependencies, so this won't be possible for me

**Comment by midudev** [comment]
Same problem for us as well. We can't keep using msw with Cypress for this problem.

**Comment by abienz** [comment]
As I am using over version 8 of NPM I can use the overrides configuration in my package.json, which is an ok workaround until this gets fixed.

https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides

**Comment by kettanaito** [comment]
Hey, @midudev. Please, pin `headers-polyfill` with Yarn resolutions or NPM overrides until the problem is resolved. Rolling back as far as one patch version should be enough.

I believe the issue is caused by me changing the `target` in `tsconfig.json`. Looks like the compiled code doesn't transpile symbols on the class, producing a syntax invalid in some contexts. 

If anyone has time to change `target` and open a pull request, I'd be thankful. Then we could merge the fix faster. Otherwise, I'll get to this when I have time.

**Comment by kettanaito** [comment]
This should be fixed in [3.0.10](https://github.com/mswjs/headers-polyfill/releases/tag/v3.0.10). Thank you, @abienz for a quick fix!

**Comment by abienz** [comment]
Just tested in my environment, and MSW appears to pickup the latest, and in turn storybook runs now for me.
